src/eval/cache.rs

```diff
@@ -1,10 +1,11 @@
-use std::fmt::Debug;
-
 use super::{
     lazy::{BlackholedError, Thunk, ThunkState, ThunkUpdateFrame},
-    IdentKind,
+    Closure, Environment, IdentKind,
+};
+use crate::{
+    identifier::Ident,
+    term::{record::FieldDeps, BindingType, RichTerm},
 };
-use crate::{eval::Closure, identifier::Ident, term::BindingType};
 
 pub trait Cache: Clone {
     type UpdateIndex; // Temporary: we won't really need that once an alternative caching mechanism gets implemented
```

Comment:
Is this alternative caching mechanism implemented in this pull request?

Reply:
A test version was implemented in this branch but wasn't included in the PR